esp/esp/program/modules/handlers/teacherbigboardmodule.py

```diff
@@ -0,0 +1,206 @@
+             hours_stats[1]),
+            ("teachers checked in today",
+             self.num_checked_in_teachers(prog)),
+        ]
+
+        numbers = [(desc, num) for desc, num in numbers if num]
+
+        timess = [
+            ("number of registered classes", self.reg_classes(prog)),
+            ("number of approved classes", self.app_classes(prog)),
+            ("number of teachers teaching", self.teach_times(prog)),
+        ]
+
+        graph_data, start = BigBoardModule.make_graph_data(timess)
+
+        hourss = self.get_hours(prog)
+
+        hours_data = self.make_hours_data(hourss)
+
+        context = {
+            "type": "Teacher",
+            "numbers": numbers,
+            "first_hour": start,
+            "graph_data": graph_data,
+            "hours_data": hours_data,
+            "loads": zip([1, 5, 15], self.load_averages()),
+        }
+        return render_to_response('program/modules/bigboardmodule/bigboard.html',
+                                  request, context)
+
+    # Numbers computed for the big board are below.  They're cached for 105
+    # seconds, which is long enough that they hopefully won't get recomputed a
+    # bunch if multiple admins are loading the page, but short enough that each
+    # time the page refreshes for the same admin, they will get new numbers
+
+    @cache_function_for(105)
+    def num_teachers_teaching(self, prog):
+        # Querying for SRs and then extracting the users saves us joining the
+        # users table.
+        return ClassSubject.objects.filter(parent_program=prog
+        ).values_list('teachers').distinct().count()
+
+    @cache_function_for(105)
+    def num_active_users(self, prog, minutes=10):
+        recent = datetime.datetime.now() - datetime.timedelta(0, minutes * 60)
+        return ClassSubject.objects.filter(parent_program=prog, timestamp__gt=recent
+        ).values_list('teachers').distinct().count()
+
+    @cache_function_for(105)
+    def num_checked_in_teachers(self, prog):
+        return Record.objects.filter(program=prog, event='teacher_checked_in').count()
+
+    @cache_function_for(105)
+    def num_class_reg(self, prog):
+        return ClassSubject.objects.filter(parent_program=prog).values_list('id').count()
+
+    @cache_function_for(105)
+    def num_class_app(self, prog):
+        return ClassSubject.objects.filter(parent_program=prog, status__gt=0, sections__status__gt=0).count()
+
+    @cache_function_for(105)
+    def reg_classes(self, prog):
+        class_times = dict(ClassSubject.objects.filter(parent_program=prog
+        ).values_list('id').annotate(Min('timestamp')))
+        return sorted(class_times.itervalues())
+
+    @cache_function_for(105)
+    def app_classes(self, prog):
+        #all ClassSubjects that are approved (and have an approved section)
+        class_times = dict(ClassSubject.objects.filter(parent_program=prog, status__gt=0, sections__status__gt=0
+        ).values_list('id').annotate(Min('timestamp')))
```

Comment:
(if you do that you may need a `distinct()` here before the `values_list` due to the filter on `sections`